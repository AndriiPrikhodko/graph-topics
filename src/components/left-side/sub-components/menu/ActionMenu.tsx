import ActionOption from './ActionOption'
import { actionList } from './ActionList'
import { useDispatch } from 'react-redux'
import { setLinkType } from '../../../../reducers/common.reducer'
import './Menu.css'

const ActionMenu: React.FC = () => {
    const dispatch = useDispatch()
    const handleChange:React.ChangeEventHandler<HTMLSelectElement> = 
    (event) => {
        dispatch(setLinkType(event.target.value));
      };

    return (
    <div>
        <select onChange={handleChange}>
            <option value="from">From</option>
            <option value="chain">Chain</option>
            <option value="bijection">Bijection</option>
            <option value="into">Into</option>
        </select>
        {actionList.map(action => {
            return <ActionOption 
                label={action.label}
                placeholder={action.placeholder}
                actionFunction={action.handleName}
                />
        })}
    </div>
    )
}

export default ActionMenu