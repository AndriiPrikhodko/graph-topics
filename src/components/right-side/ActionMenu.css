.action-wrapper-header {
    flex-shrink: 0;
    position: fixed;
    background-color: dimgray;
    display: flex;
    width: 140px;
    outline: solid;
    padding: 3px 3px;
    flex-wrap: wrap;
    align-content: center;
    border-radius: 16px;
    justify-content: space-between;
    align-items: center;
    flex-direction: row;
    margin-left: 280px;
    outline-width: 3px;
    outline-color: lightgray;
}