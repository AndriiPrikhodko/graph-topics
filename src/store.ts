import { configureStore } from '@reduxjs/toolkit'
import rootReducer from './reducers'
import thunk from 'redux-thunk'

const store = configureStore({
  reducer: {
    graphData: rootReducer.graphSlice,
    graphList: rootReducer.graphListSlice
  },
  middleware: [thunk],
})

export default store;