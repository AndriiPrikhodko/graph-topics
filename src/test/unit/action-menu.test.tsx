import {  test } from 'vitest'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ActionMenu from '../../components/left-side/sub-components/menu/ActionMenu'
import store from '../../store';
import { Provider } from 'react-redux';

test('action menu', () => {
  const menuRender = render(
    <Provider store={store}>
     <ActionMenu />
    </Provider>
  )
  
  const dataInputs = menuRender.baseElement.querySelectorAll('.data-input')
  dataInputs.forEach(inputEl => {
    expect(inputEl).toBeInTheDocument()
    expect(inputEl).toMatchSnapshot()
  })
});