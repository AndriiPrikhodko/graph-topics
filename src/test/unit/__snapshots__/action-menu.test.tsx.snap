// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`action menu data input items 1`] = `
<div
  class="data-input menu-item"
  data-testid="addGraphEdge"
>
  <label
    class="unselectable"
  >
    <div
      class="icon-encircle"
    >
      <svg
        class="action-icon"
        fill="currentColor"
        height="1em"
        id="icon-addGraphEdge"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          link nodes
        </title>
        <path
          d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        />
      </svg>
    </div>
    <input
      class="edge-input"
      data-testid="action-addGraphEdge"
      placeholder="comma separated strings"
      type="text"
    />
  </label>
</div>
`;

exports[`action menu data input items 2`] = `
<div
  class="data-input menu-item"
  data-testid="removeGraphEdge"
>
  <label
    class="unselectable"
  >
    <div
      class="icon-encircle"
    >
      <svg
        class="action-icon"
        fill="currentColor"
        height="1em"
        id="icon-addGraphEdge"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          remove edge
        </title>
        <path
          d="M304.083 405.907c4.686 4.686 4.686 12.284 0 16.971l-44.674 44.674c-59.263 59.262-155.693 59.266-214.961 0-59.264-59.265-59.264-155.696 0-214.96l44.675-44.675c4.686-4.686 12.284-4.686 16.971 0l39.598 39.598c4.686 4.686 4.686 12.284 0 16.971l-44.675 44.674c-28.072 28.073-28.072 73.75 0 101.823 28.072 28.072 73.75 28.073 101.824 0l44.674-44.674c4.686-4.686 12.284-4.686 16.971 0l39.597 39.598zm-56.568-260.216c4.686 4.686 12.284 4.686 16.971 0l44.674-44.674c28.072-28.075 73.75-28.073 101.824 0 28.072 28.073 28.072 73.75 0 101.823l-44.675 44.674c-4.686 4.686-4.686 12.284 0 16.971l39.598 39.598c4.686 4.686 12.284 4.686 16.971 0l44.675-44.675c59.265-59.265 59.265-155.695 0-214.96-59.266-59.264-155.695-59.264-214.961 0l-44.674 44.674c-4.686 4.686-4.686 12.284 0 16.971l39.597 39.598zm234.828 359.28l22.627-22.627c9.373-9.373 9.373-24.569 0-33.941L63.598 7.029c-9.373-9.373-24.569-9.373-33.941 0L7.029 29.657c-9.373 9.373-9.373 24.569 0 33.941l441.373 441.373c9.373 9.372 24.569 9.372 33.941 0z"
        />
      </svg>
    </div>
    <input
      class="edge-input"
      data-testid="action-removeGraphEdge"
      placeholder="comma separated strings"
      type="text"
    />
  </label>
</div>
`;

exports[`action menu data input items 3`] = `
<div
  class="data-input menu-item"
  data-testid="deleteGraphNode"
>
  <label
    class="unselectable"
  >
    <div
      class="icon-encircle"
    >
      <svg
        class="action-icon"
        fill="currentColor"
        height="1em"
        id="icon-addGraphEdge"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          delete node
        </title>
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
        />
      </svg>
    </div>
    <input
      class="edge-input"
      data-testid="action-deleteGraphNode"
      placeholder="node name"
      type="text"
    />
  </label>
</div>
`;
